packages/web-scripts/src/Tasks/LintTask.ts

```diff
@@ -21,8 +51,30 @@ export function lintTask(task: LintTaskDesc): SpawnSyncReturns<Buffer> {
     'cjs/',
     '--ignore-pattern',
     'esm/',
+    ...(task.format ? ['--format', task.format] : []),
+    ...(task.outputFile ? ['--output-file', task.outputFile] : []),
     ...task.restOptions,
   ];
   dbg('npx args %o', args);
-  return spawn.sync(cmd, args, { stdio: 'inherit' });
+
+  const stdout = await spawn(cmd, args, { stdio: 'inherit', ...options });
+  return (stdout || '').toString();
+}
+
+async function eslintRunWithCheckstyle(task: LintTaskDesc): Promise<string> {
+  return await eslintRun(
+    {
+      ...task,
+      format: 'checkstyle',
+      outputFile: 'checkstyle-result.xml',
```

Comment:
outputFile guarantees that the file is written, even if things are throwing. That means that we don't need `allSettled`.